lua/nvim-treesitter/parsers.lua

```diff
@@ -2,14 +2,24 @@ local api = vim.api
 local ts = vim.treesitter
 
 local ft_to_parsername = {}
+local injections_aliases = {}
 
-local function update_ft_to_parsername(name, parser)
+local function update_parser(name, parser)
+  -- Update filetype to parser table.
   if type(parser.used_by) == 'table' then
     for _, ft in pairs(parser.used_by) do
       ft_to_parsername[ft] = name
     end
   end
   ft_to_parsername[parser.filetype or name] = name
+
+  -- Update injection aliases to parser table.
+  if parser.injections and parser.injections.aliases then
+    for _, alias in pairs(parser.injections.aliases) do
+      injections_aliases[alias] = name
+    end
+  end
+  injections_aliases[name] = name
```

Comment:
do we really want the same problems as with `used_by` again. We could refactor to make the mapping `ft -> lang` and `alias -> lang` directly accessible so that the are directly configurable without any metatables and additional code magic.